app/src/main/java/com/example/newsportal/di/Module.kt

```diff
@@ -49,9 +55,14 @@ val databaseModule = module {
 }
 
 val repositoryModule = module {
-    fun provideNewsRepository(apiService: NewsService, dao: NewsDao): NewsRepository {
-        return NewsRepository(apiService, dao)
+    fun provideNewsRepository(remote: IRemoteDataSource, local: ILocalDataSource): INewsRepository {

```

Comment:
you could get rid of this provide method and implement it as data sources 
single<INewsRepository> { NewsRepository(get(), get()) }